packages/aws-cdk-lib/custom-resources/lib/provider-framework/runtime/cfn-response.ts

```diff
@@ -101,4 +106,18 @@ export function safeHandler(block: (event: any) => Promise<void>) {
   };
 }
 
+export function redactDataFromPayload(payload: OnEventResponse) {
+  // Create a deep copy of the payload object
+  const redactedPayload: OnEventResponse = JSON.parse(JSON.stringify(payload));
+
+  // Redact the data in the copied payload object
+  if (redactedPayload.Data) {
+    const keys = Object.keys(redactedPayload.Data);
+    for (const key of keys) {
+      redactedPayload.Data[key] = '*****';
```

Comment:
nit:
```suggestion
      redactedPayload.Data.get(key) = '*****';
```

Reply:
I think that's python syntax? Don't think typescript supports `{}.get()` call.

Reply:
u right lol